Adding michaeldmoore-annunciator-panel to repo.json

michaeldmoore opened the pull request:
Contributing a useful panel based on SingleStat, but with enhanced display options for showing/flashing upper and lower limits and warning levels.

daniellee commented:
Thanks for submitting this!

Here's my feedback so far.

### Meta Data

All the necessary files are in place and the plugin id is in the correct format. 

Maybe the readme could be a bit longer and a few screenshots would be great. A link to the GitHub project in the plugin.json would be nice too. Though these are not showstoppers, we can publish first and fix later if you want.

### Plugin

- You have a bug [here](https://github.com/michaeldmoore/michaeldmoore-annunciator-panel/blob/master/src/module.js#L68) so the defaults never get set. `panelDefaults` is a local variable so remove the `this`.
- Rather than logging to the console, when you get an error (like the multiple series returned by the query error). Use the AlertSrv directive and inject it via your constructor:
    ```
    constructor($scope, $injector, alertSrv) {
      this.alertSrv = alertSrv;
      ....
    ```
    Then use the `set` function to send toaster notifications to the Grafana GUI, the parameters are `title, text, severity, timeout`. Severity can be `success`, `error`, `warning` or `info`. Timeout is how many seconds it should be shown for before fading away.
    `this.alertSrv.set("Your error", message, 'error', timeout);` 
- The defaults for prefix and postfix should probably be empty. It looks a bit strange with "PRE 34% POST"
- The imports for plot and jQuery seem to be missing:
    ```
    import $ from 'jquery';
    import 'jquery.flot';
    import 'jquery.flot.pie';
    ```

michaeldmoore commented:
Thank you Daniel - hold off then until I have had a chance to address these
issues.

On Mon, Jul 10, 2017 at 5:38 AM, Daniel Lee <notifications@github.com>
wrote:

> Thanks for submitting this!
>
> Here's my feedback so far.
> Meta Data
>
> All the necessary files are in place and the plugin id is in the correct
> format.
>
> Maybe the readme could be a bit longer and a few screenshots would be
> great. A link to the GitHub project in the plugin.json would be nice too.
> Though these are not showstoppers, we can publish first and fix later if
> you want.
> Plugin
>
>    - You have a bug here
>    <https://github.com/michaeldmoore/michaeldmoore-annunciator-panel/blob/master/src/module.js#L68>
>    so the defaults never get set. panelDefaults is a local variable so
>    remove the this.
>    - Rather than logging to the console, when you get an error (like the
>    multiple series returned by the query error). Use the AlertSrv directive
>    and inject it via your constructor:
>
>    constructor($scope, $injector, alertSrv) {
>      this.alertSrv = alertSrv;
>      ....
>
>    Then use the set function to send toaster notifications to the Grafana
>    GUI, the parameters are title, text, severity, timeout. Severity can
>    be success, error, warning or info. Timeout is how many seconds it
>    should be shown for before fading away.
>    this.alertSrv.set("Your error", message, 'error', timeout);
>    - The defaults for prefix and postfix should probably be empty. It
>    looks a bit strange with "PRE 34% POST"
>    - The imports for plot and jQuery seem to be missing:
>
>    import $ from 'jquery';
>    import 'jquery.flot';
>    import 'jquery.flot.pie';
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/139#issuecomment-314092803>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADjVrkJ35IpWbMEjDz0QwIb4oC3DRvxzks5sMhsugaJpZM4OMzXK>
> .
>



-- 
*"That's not grey - I've just come from San Francisco with some flour in my
hair"*

michaeldmoore commented:
OK, version 1.0.1 is all ready.

I added a reference to 1.0.1 in repo.json, alongside the earlier version 1.0.0 version.  I'm not sure if there is any point to keep the non-released 1.0.0 version reference though.  Please advise, and/or remove.


Also - I don't think I need any reference to jquery.flot.pie, unless I'm being really thick.  I added the jquery.flot import as you suggested, as this is used by the SparkLine feature.


The setValues(data) function (copied, shamelessly from SingleStat) ends with some code that I don't understand:

```
            // Add $__name variable for using in prefix or postfix
            data.scopedVars = _.extend({}, this.panel.scopedVars);
            data.scopedVars["__name"] = {
                value: this.series[0].label
            };
        }
    }
```

Does this do anything useful?  Maybe for the SparkLine?

Anyway, thank you for everything.
Michael Moore

michaeldmoore commented:
I refreshed the files in the dist folder and updated the commit hash in repo.json as a result.

the hash for version 1.0.1 should be 542707e504b1461bb6fb70b86c8a637f93724181

daniellee commented:
As the plugin is not published yet, there is no need to increment the version number. We can change it back to 1.0.0 if you want.

The code for $__name is this [feature](https://github.com/grafana/grafana/issues/8563) where you can use the series name in the pre- or post-fix fields. I will test again later today and hopefully publish then.

michaeldmoore commented:
Any update on this yet?  was there a problem?

On Tue, Jul 11, 2017 at 3:24 AM, Daniel Lee <notifications@github.com>
wrote:

> As the plugin is not published yet, there is no need to increment the
> version number. We can change it back to 1.0.0 if you want.
>
> The code for $__name is this feature
> <https://github.com/grafana/grafana/issues/8563> where you can use the
> series name in the pre- or post-fix fields. I will test again later today
> and hopefully publish then.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/139#issuecomment-314401898>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADjVrrTi8Zm08DAHBdh77FRo1UraIt2iks5sM01hgaJpZM4OMzXK>
> .
>



-- 
*"That's not grey - I've just come from San Francisco with some flour in my
hair"*

daniellee commented:
Sorry for the delay. Have been on vacation. Will try and publish the plugin this week.

michaeldmoore commented:
No problem, hope you had a great vacation - I think you deserved some time
off too :)

On Mon, Jul 24, 2017 at 11:18 AM, Daniel Lee <notifications@github.com>
wrote:

> Sorry for the delay. Have been on vacation. Will try and publish the
> plugin this week.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/139#issuecomment-317510006>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADjVrsg3k3fvmkrXQ50SUotxhvwP_jKkks5sRN_0gaJpZM4OMzXK>
> .
>



-- 
*"That's not grey - I've just come from San Francisco with some flour in my
hair"*

daniellee commented:
Published! Thanks for your patience!

https://grafana.com/plugins/michaeldmoore-annunciator-panel
